### PR TITLE
Openpifpaf and OFM face-mask classifier TensorRT support for Jetson TX2 with Jetpack 4.4

### PR DESCRIPTION
**Pull request by undefined-references**
Changes:
* Openpifpaf_tensorrt added as a detector for Jetson and is tested on Jetson TX2.
* OFM tensorrt added as a face-mask classifier for Jetson and is tested on Jetson TX2.
* Dockerfile modified to nvcr base image 'l4t-pytorch:r32.4.4-pth1.6-py3' which is compatible with jetson's with Jetpack 4.4
* Previous dockerfile (compatible with Jetpack 4.3) **is removed** (with an assumption to have the latest versions compatibility on master, you can create a tag current state for Jetpack 4.3 and then merge this PR).
* Due to some conflicts between openpifpaf 0.12a4 and pytorch 1.6, exporting ONNX model from openpifpaf on Jetson device was not possible, so I have sent this [PR](https://github.com/neuralet/neuralet-models/pull/15) to Neuralet-Models containing openpifpaf version 0.12a4 ONNX models and I need to modify model download url in [generate_tensorrt script](https://github.com/undefined-references/smart-social-distancing/blob/8e440df64682991df118b391da0ed384d66d922c/generate_tensorrt.bash#L18) after merging that PR (before merging this PR).

### CODE REVIEW

File: generate_tensorrt.bash [hunk 1]
```diff
@@ -1,25 +1,41 @@
 #!/bin/bash
 config="$1"
+run_on_jetson="$2"
```
Comment (JsonSadler) [inline generate_tensorrt.bash:3]: Make sure this is also changed in the docs and readme

**Review by JsonSadler — APPROVED** [review]
Thanks for your great work :)

File: libs/classifiers/classifier.py [hunk 2]
```diff
@@ -1,17 +1,24 @@
+import logging
+
+logging.getLogger().setLevel(logging.INFO)
+
 class Classifier:
```
Comment (mrn-mln) [inline libs/classifiers/classifier.py:6]: Could you add the docstring for the class?

**Review by mrn-mln — APPROVED** [review]

File: generate_tensorrt.bash [hunk 3]
```diff
@@ -1,25 +1,41 @@
...
-if [ ! -f $tensorrtName ]; then
-	echo ############## Generating TensorRT Engine ######################
-	onnx2trt $onnxName -o $tensorrtName -d $precision -b 1
-fi	
+if [ ! -f $tensorrt_name_openpifpaf ]; then
+    echo "############## Generating TensorRT Engine for openpifpaf ######################"
+    onnx2trt $onnx_name_openpifpaf -o $tensorrt_name_openpifpaf -d $precision -b 1
+fi
+
+tensorrt_name_face_mask="/repo/data/tensorrt/ofm_face_mask_d${precision}.trt"
```
Comment (pgrill) [inline generate_tensorrt.bash:35]: Is correct to use the same precision in the detector and the classifier?
Maybe we can add the `TensorrtPrecision` attribute in the classifier section.

File: libs/detectors/jetson/openpifpaf_tensorrt/openpifpaf_tensorrt.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
+import time
```
Comment (pgrill) [inline libs/detectors/jetson/openpifpaf_tensorrt/openpifpaf_tensorrt.py:1]: Is this file a copy of the x86 version?

File: libs/classifiers/classifier.py (hunk 2 above)
Comment (undefined-references) [reply libs/classifiers/classifier.py:6]: Sure, thanks.

File: config-jetson.ini [hunk 5]
```diff
@@ -64,13 +64,21 @@ DailyReportTime = 06:00
...
 Device = Jetson
 ;  Name can be ssd_mobilenet_v2_coco which is trained on coco, or
-; ssd_mobilenet_v2_pedestrian_softbio trained by neuralet on softbio dataset
-Name = ssd_mobilenet_v2_coco
-;ImageSize should be 3 numbers seperated by commas, no spaces: 300,300,3
-ImageSize = 300,300,3
+; ssd_mobilenet_v2_pedestrian_softbio trained by neuralet on softbio dataset or it can be openpifpaf_tensorrt which is a pose estimation model
+Name = openpifpaf_tensorrt
+;ImageSize should be 3 numbers seperated by commas, no spaces
+ImageSize = 321,193,3
```
[the remark this answers is not in the document]
Comment (undefined-references) [reply config-jetson.ini:70]: It's a good idea, thanks.

File: generate_tensorrt.bash (hunk 3 above)
Comment (undefined-references) [reply generate_tensorrt.bash:35]: You are right. I will modify this.

File: libs/detectors/jetson/openpifpaf_tensorrt/openpifpaf_tensorrt.py (hunk 4 above)
Comment (undefined-references) [reply libs/detectors/jetson/openpifpaf_tensorrt/openpifpaf_tensorrt.py:1]: x86? no but yes for x86-gpu-tensorrt
This happens a lot on other places too due to the policy to separate modules based on platforms to have a straightforward structure. There is a solution to change the structure and add one gpu-tensorrt device for both x86-gpu-tensorrt and Jetson devices.

File: generate_tensorrt.bash (hunk 1 above)
Comment (undefined-references) [reply generate_tensorrt.bash:3]: Sure, Thanks

**Review by pgrill — APPROVED** [review]
Great work!!!

**Comment by undefined-references** [comment]
@pgrill Thanks :)

**Pull request merged**